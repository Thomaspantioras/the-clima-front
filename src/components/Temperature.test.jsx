
import Temperature from "./Temperature"

it("renders correctly",() => {
 const wrapper = shallow(
   <Temperature temp={10} city='Toronto' toggleForecast={() => {}} />
 )

expect(wrapper).toMatchSnapshot()
})
it("renders correctly again",() => {
  // render return the actual html
 const wrapper = render(
   <Temperature temp={10} city='Toronto' toggleForecast={() => {}} />
 )

expect(wrapper).toMatchSnapshot()
})

it('formats the temp correctly', () => {
  const wrapper = mount(<Temperature temp={10} city='Toronto' toggleForecast={() => {}} />)

  const text = wrapper.find("LargeText").text();
  console.log(text)
  expect(text).toEqual("10Â°c")
});

it('calls toggleForecast on Click', () => {
  const spy = sinon.spy();
  const wrapper = mount(<Temperature temp={10} city='Toronto' toggleForecast={spy} />)
  
  wrapper.find('div').first().simulate("click")

  expect(spy.calledOnce).toBe(true)
});