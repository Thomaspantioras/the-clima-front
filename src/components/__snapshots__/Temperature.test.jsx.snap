// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(styled.div)
  onClick={[Function]}
>
  <LargeText>
    10
    °c
  </LargeText>
  <SmallText>
    Toronto
  </SmallText>
</Styled(styled.div)>
`;

exports[`renders correctly again 1`] = `
<div
  class="sc-bwzfXH ilYHYa sc-bdVaJa dQdFv"
>
  <span
    class="sc-ifAKCX XBTtJ"
  >
    10°c
  </span>
  <span
    class="sc-EHOje exztiK"
  >
    Toronto
  </span>
</div>
`;
